slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbHealthCheck.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DbHealthCheck implements Runnable {
+
+  private final SlashingProtectionContext slashingProtectionContext;
+  private final long dbHealthCheckTimeoutMilliseconds;
+  private final AtomicBoolean isDbUp;
+
+  public DbHealthCheck(
+      final SlashingProtectionContext slashingProtectionContext,
+      final long dbHealthCheckTimeoutMilliseconds) {
+    this.slashingProtectionContext = slashingProtectionContext;
+    this.dbHealthCheckTimeoutMilliseconds = dbHealthCheckTimeoutMilliseconds;
+    this.isDbUp = new AtomicBoolean(true);
+  }
+
+  @Override
+  public void run() {
+    try {
+      CompletableFuture<Integer> future =
+          CompletableFuture.supplyAsync(
+              () ->
+                  slashingProtectionContext
+                      .getSlashingProtectionJdbi()
+                      .withHandle(h -> h.execute("SELECT 1")));
+      // Check db health with timeout.
+      future.get(this.dbHealthCheckTimeoutMilliseconds, TimeUnit.MILLISECONDS);
+      isDbUp.set(true);
+    } catch (Exception e) {
```

Comment:
Wondering if this should be `catch (Throwable e)` as it looks like jdbi could throw a RuntimeException and we don't want to kill the health check thread in this case.

Reply:
Updated